### PR TITLE
Add rollback function. To be used when exceptions happen and transact…

### PR DESCRIPTION
**Pull request by melmccann**
…ions are left open.

### CODE REVIEW

File: Components/DB/Dbal/DbalStepsProvider.php [hunk 1]
```diff
@@ -291,4 +291,12 @@ protected function getConnection($context)
 
         return $this->doctrine->getConnection($connectionName);
     }
+
+    /**
+     * @param $context
+     */
+    protected function rollback(&$context)
+    {
+        $this->getConnection($context)->executeQuery('ROLLBACK', [], []);
```
Comment (campru) [inline Components/DB/Dbal/DbalStepsProvider.php]: Doctrine DBAL has a method for the rollback. https://www.doctrine-project.org/projects/doctrine-dbal/en/2.8/reference/transactions.html

File: Components/DB/Dbal/DbalStepsProvider.php [hunk 2]
```diff
@@ -291,4 +291,12 @@ protected function getConnection($context)
 
         return $this->doctrine->getConnection($connectionName);
     }
+
+    /**
+     * @param $context
+     */
+    protected function rollback(&$context)
```
Comment (campru) [inline Components/DB/Dbal/DbalStepsProvider.php]: In this case you don't need to pass the context as reference

File: Components/DB/Dbal/DbalStepsProvider.php (hunk 1 above)
Comment (melmccann) [reply Components/DB/Dbal/DbalStepsProvider.php]: yes. I will remove from here. We will handle elsewhere.

File: Components/DB/Dbal/DbalStepsProvider.php (hunk 2 above)
Comment (melmccann) [reply Components/DB/Dbal/DbalStepsProvider.php]: will remove this. Thanks @campru

**Pull request closed without merging**